app/hooks/quest.ts

```diff
@@ -110,29 +114,99 @@ export const useNavigatePrevQuest = () => {
 
     if (previousQuestIdx) {
       dispatch(questSlice.actions.popFromCurrentQuestStack());
+    } else {
+      dispatch(questSlice.actions.clearQuestStack());
     }
   }, [currentQuestIdx, dispatch, navigation, previousQuestIdx]);
 
   return navigatePrevQuest;
 };
 
-export const useParseTextWithNickname = (text: string): string => {
-  const currentNickname = useAppSelector(
-    state => state.cache.nicknames?.current,
+export const useParseTextWithNickname = ({
+  text,
+  textHasNickname,
+  preset,
+  style,
+  nicknamePreset,
+  nicknameStyle,
+}: {
+  text: string;
+  textHasNickname: boolean;
+  preset?: ExtendedTextPresets;
+  style?: TextStyle;
+  nicknamePreset?: ExtendedTextPresets;
+  nicknameStyle?: TextStyle;
+}) => {
```

Comment:
type it properly with our React.FC or ExtendedText usage, you can do it without objects, because you are returning JSX object as I see